ci-operator/step-registry/aws/provision/vpc/shared/aws-provision-vpc-shared-commands.sh

```diff
@@ -8,6 +8,11 @@ set -o pipefail
 trap 'save_stack_events_to_artifacts' EXIT TERM INT
 
 export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+if [[ ${ENABLE_SHARED_VPC} == "yes" ]]; then
+  CLUSTER_CREATOR_USER_ARN=$(aws sts get-caller-identity | jq -r '.Arn')
+  CLUSTER_CREATOR_AWS_ACCOUNT_NO=$(echo $CLUSTER_CREATOR_USER_ARN | awk -F ":" '{print $5}')
+  export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred_shared_account"
+fi
```

Comment:
Based on [the rehearsal,](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/39946/rehearse-39946-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-aws/1668629045471875072) it looks like this change might not be working:

```
[36mINFO[0m[2023-06-13T14:41:33Z] Running step installer-rehearse-aws-aws-provision-vpc-shared. 
[36mINFO[0m[2023-06-13T14:41:47Z] Logs for container test in pod installer-rehearse-aws-aws-provision-vpc-shared: 
[36mINFO[0m[2023-06-13T14:41:47Z] 
Unable to locate credentials. You can configure credentials by running "aws configure".

Unable to locate credentials. You can configure credentials by running "aws configure".
{"component":"entrypoint","error":"wrapped process failed: exit status 253","file":"k8s.io/test-infra/prow/entrypoint/run.go:79","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2023-06-13T14:41:47Z"}
error: failed to execute wrapped command: exit status 253 
[36mINFO[0m[2023-06-13T14:41:47Z] Step installer-rehearse-aws-aws-provision-vpc-shared failed after 13s. 
[36mINFO[0m[2023-06-13T14:41:47Z] Step phase pre failed after 44s.    
```